src/domain/bitcoin/onchain/index.types.d.ts

```diff
@@ -64,6 +64,13 @@ type PayToAddressArgs = {
   targetConfirmations: TargetConfirmations
 }
 
+type IncomingOnChainTxHandler = {
+  balanceByAddress(): { [key: OnChainAddress]: BtcPaymentAmount } | ValidationError
```

Comment:
shouldn't this function take an arg? ie:

```suggestion
  balanceByAddress(address: OnchainAddress): { [key: OnChainAddress]: BtcPaymentAmount } | ValidationError
```

Reply:
otherwise maybe it should be `balanceByAddresses`?

Reply:
It uses the `txns` array internal to the `IncomingOnChainTxHandler`. Changed the name here: https://github.com/GaloyMoney/galoy/pull/1512/commits/7267097055d3f68eb8ebb24bb0e32ae4b178bc41

Reply:
I actually prefer `balanceByAddresses` than `balanceByTxnsAddresses`. the thing that was "bothering" me in the previous commit is the use of singular where a list were returned.

Reply:
I can change in a follow-up PR 👍🏾

Reply:
Addressed: #1630